### ISSUE TITLE
used out of scope - Triggered when using ES6 anonymous functions

### ISSUE THREAD

**Issue by dani-h** [jshint/jshint#2028]
I'm using the esnext option.

```javascript
for(;;) {
  var cmp = {}
}

var cmp_ids = component.children.map((cmp) => {
  return cmp.id
 })
```

If I change the code to

```javascript
for(;;) {
  var cmp = {}
}

var cmp_ids = component.children.map(function(cmp) {
  return cmp.id
 })
```

It no longer warns.

**Comment by jugglinmike** [comment]
We resolved this with gh-1989, and [the fix was released in version 2.6.0](https://github.com/jshint/jshint/releases/tag/2.6.0) back in January of this year. Thanks for the report!

**Comment by dani-h** [comment]
:)
